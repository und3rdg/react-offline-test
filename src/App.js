import React from 'react';
import axios from 'axios'
import {fuel_names, fuel_perc} from './helpers'
import {Legend} from './Legend'
import Pie_chart from './Pie_chart'

export default class App extends React.Component {
    constructor(props){
        super(props)
        this.state = {
            data:{},
            mode: {
                dev: !!JSON.parse(window.sessionStorage.getItem('fake_api')),
                test: !!props.test,
            },
            url: {
                api: 'https://api.carbonintensity.org.uk/generation',
            },
            colors: [
                "#dce8bf",
                "#bfe8d7",
                "#bfcce8",
                "#e0bfe8",
                "#ff8181",
                "#ffde81",
                "#acff81",
                "#81f3ff",
                "#e081ff",
                "#ffdede",
                "#b4d422",
                "#dee7ff",
                "#deffe8",
                "#fffcde",
                "#e8bfbf",
            ]
        }

        const {mode, url} = this.state
        if(!mode.test) {
            axios.get(url.api)
                .then(res=>{
                    this.setState(res.data)
                })
                .catch(err=>console.error(err))
        }

    }
    
    render() {
        const data = !!this.state.data && this.state.data
        const names = fuel_names(data.generationmix)
        const perc = fuel_perc(data.generationmix)
        const colors = this.state.colors

        const style = {
            wrap: {
            display: "grid",
            gridTemplateColumns: "200px 200px",
                gridGap: "10px",
            },
            title: {
                gridColumn: "1 / 3",
            }
        } 
        return (
            <div style={style.wrap}>
                <h1 style={style.title}>UK Energy Mix</h1>
                {!!names.length && <Legend  names={names} colors={colors} />}
                {!!names.length && <Pie_chart perc={perc} colors={colors} />}
            </div>
        )
    }
}
