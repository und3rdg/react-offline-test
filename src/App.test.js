import React from 'react';
import App from './App';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
    const tree = renderer
        .create(<App test={true} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
