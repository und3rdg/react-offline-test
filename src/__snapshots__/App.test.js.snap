// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  style={
    Object {
      "display": "grid",
      "gridGap": "10px",
      "gridTemplateColumns": "200px 200px",
    }
  }
>
  <h1
    style={
      Object {
        "gridColumn": "1 / 3",
      }
    }
  >
    UK Energy Mix
  </h1>
</div>
`;
